plugins/mana/plugin.go

```diff
@@ -371,8 +408,253 @@ func GetPendingMana(value float64, n time.Duration) float64 {
+					baseManaVector, _ := mana.NewBaseManaVector(mana.ConsensusMana)
+					cbmvPast = baseManaVector.(*mana.ConsensusBaseManaVector)
+					return false
+				}
+				return true
+			})
+		}
+	}
+
+	var eventLogs []mana.Event
+	consensusEventsLogStorage.ForEach(func(key []byte, cachedObject objectstorage.CachedObject) bool {
+		cachedPe := &mana.CachedPersistableEvent{CachedObject: cachedObject}
+		defer cachedPe.Release()
+		pe := cachedPe.Unwrap()
+		if pe.Time.After(t) {
+			return true
+		}
+
+		// already consumed in stored base mana vector.
+		if cachedMetadata.Exists() && cbmvPast.Size() > 0 {
+			metadata := cachedMetadata.Unwrap()
+			if pe.Time.Before(metadata.Timestamp) {
+				return true
+			}
+		}
+
+		var ev mana.Event
+		ev, err = mana.FromPersistableEvent(pe)
+		if err != nil {
+			return false
+		}
+		eventLogs = append(eventLogs, ev)
+		return true
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	sort.Slice(eventLogs, func(i, j int) bool {
+		var timeI, timeJ time.Time
+		var typeI, _ byte
+		switch eventLogs[i].Type() {
+		case mana.EventTypePledge:
+			timeI = eventLogs[i].(*mana.PledgedEvent).Time
+			typeI = mana.EventTypePledge
+		case mana.EventTypeRevoke:
+			timeI = eventLogs[i].(*mana.RevokedEvent).Time
+			typeI = mana.EventTypeRevoke
+		}
+
+		switch eventLogs[j].Type() {
+		case mana.EventTypePledge:
+			timeJ = eventLogs[j].(*mana.PledgedEvent).Time
+			_ = mana.EventTypePledge
+		case mana.EventTypeRevoke:
+			timeJ = eventLogs[j].(*mana.RevokedEvent).Time
+			_ = mana.EventTypeRevoke
+		}
+
+		if !timeI.Equal(timeJ) {
+			return timeI.Before(timeJ)
+		}
+
+		return typeI == mana.EventTypeRevoke
+	})
+	err = cbmvPast.BuildPastBaseVector(eventLogs, t)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	err = cbmvPast.UpdateAll(t)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cbmvPast, eventLogs, nil
+}
+
+func checkConsensusEventStorage() {
+	size := consensusEventsLogStorage.GetSize()
+	if size < maxConsensusEventsInStorage {
+		return
+	}
+	t := time.Now()
+	cbmvPast, logs, err := GetPastConsensusManaVector(t)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	// store cbmv
+	if err = consensusBaseManaPastVectorStorage.Prune(); err != nil {
+		log.Errorf("error pruning consensus base mana vector storage: %w", err)
+		return
+	}
+	for _, p := range cbmvPast.ToPersistables() {
+		consensusBaseManaPastVectorStorage.Store(p).Release()
+	}
+
+	//store the metadata
+	metadata := &mana.ConsensusBasePastManaVectorMetadata{
+		Timestamp: t,
+	}
+
+	cachedObj := consensusBaseManaPastVectorMetadataStorage.Get([]byte(mana.ConsensusBaseManaPastVectorMetadataStorageKey))
+	cachedMetadata := &mana.CachedConsensusBasePastManaVectorMetadata{CachedObject: cachedObj}
+	defer cachedMetadata.Release()
+
+	if !cachedMetadata.Exists() {
+		consensusBaseManaPastVectorMetadataStorage.Store(metadata).Release()
+	} else {
+		m := cachedMetadata.Unwrap()
+		m.Update(metadata)
+	}
+
+	err = writeEventsToCSV(logs)
```

Comment:
Yes, we could do it like this. Or an easier (and real time) approach is to have a separate background worker listening to `pledge/revoke` events, gathering them for couple seconds, and writing them to the file if there are things to write. Then we don't need too have the logic intertwined with the event storage. We can even have it as a separate extension plugin.